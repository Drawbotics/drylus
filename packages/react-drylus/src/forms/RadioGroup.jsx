import React from 'react';
import v4 from 'uuid/v4';
import { css, cx } from 'emotion';
import sv from '@drawbotics/style-vars';
import PropTypes from 'prop-types';

import Hint from './Hint';
import Icon from '../components/Icon';
import { Sizes } from '../base';


const styles = {
  radioGroup: css`
    display: inline-block;
  `,
  radioWrapper: css`
    margin-bottom: ${sv.marginSmall};
  `,
  base: css`
    position: relative;
    display: inline-block;
  `,
  wrapper: css`
    display: inline-flex;
    align-items: center;
    justify-content: flex-start;
    cursor: pointer;

    &:hover {
      [data-element="sprite"] {
        background: ${sv.neutralDark};
      }
    }
  `,
  disabled: css`
    cursor: not-allowed !important;

    > label {
      cursor: not-allowed !important;
      color: ${sv.colorDisabled};
    }
  `,
  error: css`
    [data-element="sprite"] {
      box-shadow: inset 0px 0px 0px 2px ${sv.red};
    }

    [data-element="icon"] {
      background: ${sv.red};
    }
  `,
  radio: css`
    height: ${sv.defaultMargin};
    width: ${sv.defaultMargin};
    position: relative;
    overflow: hidden;
  `,
  input: css`
    visibility: hidden;

    &:checked + [data-element="sprite"] {
      [data-element="icon"] {
        transform: scale(1);
        border-radius: 0;
      }
    }

    &:disabled + [data-element="sprite"] {
      cursor: not-allowed;
      background: ${sv.neutralLight} !important;

      [data-element="icon"] {
        opacity: 0.7;
      }
    }
  `,
  label: css`
    margin-left: ${sv.marginExtraSmall};
    color: ${sv.colorPrimary};

    &:hover {
      cursor: pointer;
    }
  `,
  sprite: css`
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    border-radius: 1000px;
    background: ${sv.neutral};
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    cursor: pointer;
    transition: ${sv.defaultTransition};
    overflow: hidden;
  `,
  iconLabel: css`
    cursor: inherit;
    color: ${sv.white};
    height: 100%;
    width: 100%;
    background: ${sv.green};
    line-height: ${sv.marginLarge};
    transform: scale(0);
    transition: all ${sv.transitionTimeShort} ${sv.bouncyTransitionCurve};
    border-radius: 100px;

    > i {
      font-size: 1.1rem;
    }
  `,
  small: css`
    > div {
      height: ${sv.marginSmall};
      width: ${sv.marginSmall};
    }
    [data-element="icon"] {
      line-height: calc(${sv.marginSmall} + 1px);

       > i {
         font-size: 0.7rem;
       }
    }

    [data-element="label"] {
      font-size: 0.9rem;
    }
  `,
};


const Radio = ({
  value,
  onChange,
  valueKey='value',
  labelKey='label',
  disabled,
  error,
  size,
  checked,
  children,
  ...rest,
}) => {
  const id = v4();
  return (
    <div className={styles.base}>
      <label className={cx(styles.wrapper, {
        [styles[size?.toLowerCase()]]: size,
        [styles.disabled]: disabled,
        [styles.error]: error,
      })} htmlFor={id}>
        <div className={styles.radio}>
          <input
            disabled={disabled}
            checked={checked}
            value={value}
            id={id}
            type="radio"
            className={styles.input}
            onChange={onChange}
            {...rest} />
          <div data-element="sprite" className={styles.sprite}>
            <label data-element="icon" className={styles.iconLabel} htmlFor={id}>
              <Icon bold name="check" />
            </label>
          </div>
        </div>
        {do{
          if (children) {
            <label data-element="label" className={styles.label} htmlFor={id}>
              {children}
            </label>
          }
        }}
      </label>
    </div>
  );
};



const RadioGroup = ({
  value,
  onChange,
  options=[],
  valueKey='value',
  labelKey='label',
  error,
  size,
  className,
  hint,
  ...rest,
}) => {

  const handleOnChange = (e) => {
    e.stopPropagation();
    onChange ? onChange(e.target.value, e.target.name) : null;
  };

  return (
    <div className={cx(styles.radioGroup, className)}>
      <div className={styles.radios}>
        {options.map((option) => (
          <div key={option[valueKey]} className={styles.radioWrapper}>
            <Radio
              size={size}
              error={!! error}
              onChange={handleOnChange}
              checked={value === option[valueKey]}
              value={option[valueKey]}
              disabled={option.disabled}
              {...rest}>{option[labelKey]}</Radio>
          </div>
        ))}
      </div>
      {do{
        if (error && typeof error === 'string') {
          <Hint error>{error}</Hint>
        }
        else if (hint) {
          <Hint>{hint}</Hint>
        }
      }}
    </div>
  );
};


RadioGroup.propTypes = {
  /** Determines the radio components which will be rendered */
  options: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    label: PropTypes.string.isRequired,
    disabled: PropTypes.bool,
  })),

  /** Used to pick each value in the options array */
  valueKey: PropTypes.string,

  /** Used to pick each label in the options array */
  labelKey: PropTypes.string,

  /** Triggered when radio value is changed */
  onChange: PropTypes.func,

  /** If true, none of the checkboxes are clickable */
  disabled: PropTypes.bool,

  /** Determines which value is currently active */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),

  /** Error text to prompt the user to act, or a boolean if you don't want to show a message */
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),

  /** Size of the radio buttons. Can be small */
  size: PropTypes.oneOf([Sizes.SMALL, Sizes.DEFAULT]),

  /** Passed to the wrapper component, to override any styles */
  className: PropTypes.string,

  /** Small text shown below the group, replaced by error if present */
  hint: PropTypes.string,
};


RadioGroup.defaultProps = {
  valueKey: 'value',
  labelKey: 'label',
  size: Sizes.DEFAULT,
};


export default RadioGroup;
