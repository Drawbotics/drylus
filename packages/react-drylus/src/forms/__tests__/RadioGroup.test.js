import React from 'react';
import { create } from 'react-test-renderer';

import RadioGroup from '../RadioGroup';


jest.mock('uuid/v4', () => {
  return jest.fn(() => 1);
});


describe('RadioGroup', () => {
  const onChange = jest.fn();

  describe('matches snapshot when', () => {
    it('is has no value', () => {
      const tree = create(
        <RadioGroup
          onChange={onChange}
          options={[{
            label: '1',
            value: '1',
          }, {
            label: '2',
            value: '3',
          }]} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is has a value', () => {
      const tree = create(
        <RadioGroup
          value="1"
          onChange={onChange}
          options={[{
            label: '1',
            value: '1',
          }, {
            label: '2',
            value: '3',
          }]} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is read only', () => {
      const tree = create(
        <RadioGroup
          value="1"
          options={[{
            label: '1',
            value: '1',
          }, {
            label: '2',
            value: '3',
          }]} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  
  describe('changes state when clicked', () => {
    let value = null;

    expect(value).toBeNull();

    const component = create(
      <RadioGroup
        value={value}
        options={[{
          label: '1',
          value: '1',
        }, {
          label: '2',
          value: '3',
        }]}
        onChange={(v) => value = v} />
    );

    const input = component.root.findAllByProps({ type: 'radio' })[0];
    const changeEvent = new Event('change');
    Object.defineProperty(changeEvent, 'target', {
      writable: false,
      value: { value: input.props.value },
    });

    input.props.onChange(changeEvent);

    expect(value).toEqual('1');
  });
});