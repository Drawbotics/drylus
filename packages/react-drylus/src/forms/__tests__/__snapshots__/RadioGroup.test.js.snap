// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup matches snapshot when is has a value 1`] = `
.css-cs17kl-RadioGroup__radioGroup {
  display: inline-block;
}

.css-1v7q1co-RadioGroup__radioWrapper {
  margin-bottom: 8px;
}

.css-1gkg9qh-RadioGroup__root {
  position: relative;
  display: inline-block;
}

.css-njatfr-RadioGroup__wrapper {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.css-njatfr-RadioGroup__wrapper:hover [data-element="sprite"] {
  background: #879ba6;
}

.css-a4uemc-RadioGroup__radio {
  height: calc(24px - 4px);
  width: calc(24px - 4px);
  position: relative;
  overflow: hidden;
}

.css-y0i22o-RadioGroup__input {
  visibility: hidden;
}

.css-y0i22o-RadioGroup__input:checked + [data-element="sprite"] [data-element="icon"] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
}

.css-y0i22o-RadioGroup__input:checked + [data-element="sprite"] [data-element='locked-icon'] {
  background: #3FBC9B;
  color: #fff;
}

.css-y0i22o-RadioGroup__input:disabled + [data-element="sprite"] {
  cursor: not-allowed;
  background: #eaeff4 !important;
}

.css-y0i22o-RadioGroup__input:disabled + [data-element="sprite"] [data-element="icon"] {
  opacity: 0.7;
}

.css-tva0le-RadioGroup__label {
  margin-left: 8px;
  color: #172b4e;
  position: relative;
  top: 1px;
}

.css-tva0le-RadioGroup__label:hover {
  cursor: pointer;
}

.css-w8mlgb-RadioGroup__sprite {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 1000px;
  background: #d3dce6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  overflow: hidden;
}

.css-k0mp4-RadioGroup__iconLabel {
  cursor: inherit;
  color: #fff;
  height: 100%;
  width: 100%;
  background: #3FBC9B;
  line-height: 24px;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  border-radius: 100px;
}

.css-k0mp4-RadioGroup__iconLabel > i {
  font-size: 0.8rem;
}

.css-g1tqqb-Icon__root-Icon__bold {
  color: inherit;
  font-weight: bold !important;
}

<div
  className="css-cs17kl-RadioGroup__radioGroup"
>
  <div>
    <div
      className="css-1v7q1co-RadioGroup__radioWrapper"
    >
      <div
        className="css-1gkg9qh-RadioGroup__root"
      >
        <label
          className="css-njatfr-RadioGroup__wrapper"
          htmlFor={1}
        >
          <div
            className="css-a4uemc-RadioGroup__radio"
          >
            <input
              checked={true}
              className="css-y0i22o-RadioGroup__input"
              id={1}
              onChange={[Function]}
              type="radio"
              value="1"
            />
            <div
              className="css-w8mlgb-RadioGroup__sprite"
              data-element="sprite"
            >
              <label
                className="css-k0mp4-RadioGroup__iconLabel"
                data-element="icon"
                htmlFor={1}
              >
                <i
                  className="Drycon Drycon-check css-g1tqqb-Icon__root-Icon__bold"
                />
              </label>
            </div>
          </div>
          <label
            className="css-tva0le-RadioGroup__label"
            data-element="label"
            htmlFor={1}
          >
            1
          </label>
        </label>
      </div>
    </div>
    <div
      className="css-1v7q1co-RadioGroup__radioWrapper"
    >
      <div
        className="css-1gkg9qh-RadioGroup__root"
      >
        <label
          className="css-njatfr-RadioGroup__wrapper"
          htmlFor={1}
        >
          <div
            className="css-a4uemc-RadioGroup__radio"
          >
            <input
              checked={false}
              className="css-y0i22o-RadioGroup__input"
              id={1}
              onChange={[Function]}
              type="radio"
              value="3"
            />
            <div
              className="css-w8mlgb-RadioGroup__sprite"
              data-element="sprite"
            >
              <label
                className="css-k0mp4-RadioGroup__iconLabel"
                data-element="icon"
                htmlFor={1}
              >
                <i
                  className="Drycon Drycon-check css-g1tqqb-Icon__root-Icon__bold"
                />
              </label>
            </div>
          </div>
          <label
            className="css-tva0le-RadioGroup__label"
            data-element="label"
            htmlFor={1}
          >
            2
          </label>
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioGroup matches snapshot when is has no value 1`] = `
.css-cs17kl-RadioGroup__radioGroup {
  display: inline-block;
}

.css-1v7q1co-RadioGroup__radioWrapper {
  margin-bottom: 8px;
}

.css-1gkg9qh-RadioGroup__root {
  position: relative;
  display: inline-block;
}

.css-njatfr-RadioGroup__wrapper {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.css-njatfr-RadioGroup__wrapper:hover [data-element="sprite"] {
  background: #879ba6;
}

.css-a4uemc-RadioGroup__radio {
  height: calc(24px - 4px);
  width: calc(24px - 4px);
  position: relative;
  overflow: hidden;
}

.css-y0i22o-RadioGroup__input {
  visibility: hidden;
}

.css-y0i22o-RadioGroup__input:checked + [data-element="sprite"] [data-element="icon"] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
}

.css-y0i22o-RadioGroup__input:checked + [data-element="sprite"] [data-element='locked-icon'] {
  background: #3FBC9B;
  color: #fff;
}

.css-y0i22o-RadioGroup__input:disabled + [data-element="sprite"] {
  cursor: not-allowed;
  background: #eaeff4 !important;
}

.css-y0i22o-RadioGroup__input:disabled + [data-element="sprite"] [data-element="icon"] {
  opacity: 0.7;
}

.css-tva0le-RadioGroup__label {
  margin-left: 8px;
  color: #172b4e;
  position: relative;
  top: 1px;
}

.css-tva0le-RadioGroup__label:hover {
  cursor: pointer;
}

.css-w8mlgb-RadioGroup__sprite {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 1000px;
  background: #d3dce6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  overflow: hidden;
}

.css-k0mp4-RadioGroup__iconLabel {
  cursor: inherit;
  color: #fff;
  height: 100%;
  width: 100%;
  background: #3FBC9B;
  line-height: 24px;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  border-radius: 100px;
}

.css-k0mp4-RadioGroup__iconLabel > i {
  font-size: 0.8rem;
}

.css-g1tqqb-Icon__root-Icon__bold {
  color: inherit;
  font-weight: bold !important;
}

<div
  className="css-cs17kl-RadioGroup__radioGroup"
>
  <div>
    <div
      className="css-1v7q1co-RadioGroup__radioWrapper"
    >
      <div
        className="css-1gkg9qh-RadioGroup__root"
      >
        <label
          className="css-njatfr-RadioGroup__wrapper"
          htmlFor={1}
        >
          <div
            className="css-a4uemc-RadioGroup__radio"
          >
            <input
              checked={false}
              className="css-y0i22o-RadioGroup__input"
              id={1}
              onChange={[Function]}
              type="radio"
              value="1"
            />
            <div
              className="css-w8mlgb-RadioGroup__sprite"
              data-element="sprite"
            >
              <label
                className="css-k0mp4-RadioGroup__iconLabel"
                data-element="icon"
                htmlFor={1}
              >
                <i
                  className="Drycon Drycon-check css-g1tqqb-Icon__root-Icon__bold"
                />
              </label>
            </div>
          </div>
          <label
            className="css-tva0le-RadioGroup__label"
            data-element="label"
            htmlFor={1}
          >
            1
          </label>
        </label>
      </div>
    </div>
    <div
      className="css-1v7q1co-RadioGroup__radioWrapper"
    >
      <div
        className="css-1gkg9qh-RadioGroup__root"
      >
        <label
          className="css-njatfr-RadioGroup__wrapper"
          htmlFor={1}
        >
          <div
            className="css-a4uemc-RadioGroup__radio"
          >
            <input
              checked={false}
              className="css-y0i22o-RadioGroup__input"
              id={1}
              onChange={[Function]}
              type="radio"
              value="3"
            />
            <div
              className="css-w8mlgb-RadioGroup__sprite"
              data-element="sprite"
            >
              <label
                className="css-k0mp4-RadioGroup__iconLabel"
                data-element="icon"
                htmlFor={1}
              >
                <i
                  className="Drycon Drycon-check css-g1tqqb-Icon__root-Icon__bold"
                />
              </label>
            </div>
          </div>
          <label
            className="css-tva0le-RadioGroup__label"
            data-element="label"
            htmlFor={1}
          >
            2
          </label>
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioGroup matches snapshot when is read only 1`] = `
.css-10jseif-Icon__root {
  color: inherit;
}

.css-cs17kl-RadioGroup__radioGroup {
  display: inline-block;
}

.css-1v7q1co-RadioGroup__radioWrapper {
  margin-bottom: 8px;
}

.css-1gkg9qh-RadioGroup__root {
  position: relative;
  display: inline-block;
}

.css-a4uemc-RadioGroup__radio {
  height: calc(24px - 4px);
  width: calc(24px - 4px);
  position: relative;
  overflow: hidden;
}

.css-y0i22o-RadioGroup__input {
  visibility: hidden;
}

.css-y0i22o-RadioGroup__input:checked + [data-element="sprite"] [data-element="icon"] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
}

.css-y0i22o-RadioGroup__input:checked + [data-element="sprite"] [data-element='locked-icon'] {
  background: #3FBC9B;
  color: #fff;
}

.css-y0i22o-RadioGroup__input:disabled + [data-element="sprite"] {
  cursor: not-allowed;
  background: #eaeff4 !important;
}

.css-y0i22o-RadioGroup__input:disabled + [data-element="sprite"] [data-element="icon"] {
  opacity: 0.7;
}

.css-tva0le-RadioGroup__label {
  margin-left: 8px;
  color: #172b4e;
  position: relative;
  top: 1px;
}

.css-tva0le-RadioGroup__label:hover {
  cursor: pointer;
}

.css-w8mlgb-RadioGroup__sprite {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 1000px;
  background: #d3dce6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  overflow: hidden;
}

.css-k0mp4-RadioGroup__iconLabel {
  cursor: inherit;
  color: #fff;
  height: 100%;
  width: 100%;
  background: #3FBC9B;
  line-height: 24px;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  border-radius: 100px;
}

.css-k0mp4-RadioGroup__iconLabel > i {
  font-size: 0.8rem;
}

.css-1pay92b-RadioGroup__wrapper-RadioGroup__readOnly {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
  pointer-events: none;
}

.css-1pay92b-RadioGroup__wrapper-RadioGroup__readOnly:hover [data-element="sprite"] {
  background: #879ba6;
}

.css-1pay92b-RadioGroup__wrapper-RadioGroup__readOnly [data-element="sprite"] {
  background: #eaeff4 !important;
}

.css-1pay92b-RadioGroup__wrapper-RadioGroup__readOnly [data-element="locked-icon"] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #879ba6;
}

<div
  className="css-cs17kl-RadioGroup__radioGroup"
>
  <div>
    <div
      className="css-1v7q1co-RadioGroup__radioWrapper"
    >
      <div
        className="css-1gkg9qh-RadioGroup__root"
      >
        <label
          className="css-1pay92b-RadioGroup__wrapper-RadioGroup__readOnly"
          htmlFor={1}
        >
          <div
            className="css-a4uemc-RadioGroup__radio"
          >
            <input
              checked={true}
              className="css-y0i22o-RadioGroup__input"
              id={1}
              onChange={[Function]}
              type="radio"
              value="1"
            />
            <div
              className="css-w8mlgb-RadioGroup__sprite"
              data-element="sprite"
            >
              <label
                className="css-k0mp4-RadioGroup__iconLabel"
                data-element="locked-icon"
              >
                <i
                  className="css-10jseif-Icon__root Drycon Drycon-lock"
                />
              </label>
            </div>
          </div>
          <label
            className="css-tva0le-RadioGroup__label"
            data-element="label"
            htmlFor={1}
          >
            1
          </label>
        </label>
      </div>
    </div>
    <div
      className="css-1v7q1co-RadioGroup__radioWrapper"
    >
      <div
        className="css-1gkg9qh-RadioGroup__root"
      >
        <label
          className="css-1pay92b-RadioGroup__wrapper-RadioGroup__readOnly"
          htmlFor={1}
        >
          <div
            className="css-a4uemc-RadioGroup__radio"
          >
            <input
              checked={false}
              className="css-y0i22o-RadioGroup__input"
              id={1}
              onChange={[Function]}
              type="radio"
              value="3"
            />
            <div
              className="css-w8mlgb-RadioGroup__sprite"
              data-element="sprite"
            >
              <label
                className="css-k0mp4-RadioGroup__iconLabel"
                data-element="locked-icon"
              >
                <i
                  className="css-10jseif-Icon__root Drycon Drycon-lock"
                />
              </label>
            </div>
          </div>
          <label
            className="css-tva0le-RadioGroup__label"
            data-element="label"
            htmlFor={1}
          >
            2
          </label>
        </label>
      </div>
    </div>
  </div>
</div>
`;
